demos/profiling/profiling_container/src/demo.py

```diff
@@ -0,0 +1,243 @@
+import argparse
+import sys
+import os
+import time
+
+import norfair
+from norfair import Tracker, Video
+
+sys.path.append("/pose/src")
+
+from utils import (
+    get_model,
+    get_preprocesser,
+    get_postprocesser,
+    get_distance_function,
+    get_filter_setup,
+    DISTANCE_THRESHOLD,
+    DETECTION_THRESHOLD,
+    HIT_COUNTER_MAX,
+    INITIALIZATION_DELAY,
+    POINTWISE_HIT_COUNTER_MAX,
+    model_path,
+    model_resolution,
+)
+
+
+def process_video(
+    video,
+    detector,
+    tracker,
+    preprocesser,
+    postprocesser,
+    frame_skip_period=1,
+    create_video=False,
+):
+
+    tracker_time_mean = 0
+    detector_time_mean = 0
+
+    start_time = time.time()
+
+    total_frames = 0
+    for frame_number, frame in enumerate(video):
+        total_frames += 1
+
+        if frame_number % frame_skip_period == 0:
+
+            start_frame_time = time.time()
+            data = preprocesser(frame)
+            cmap, paf = detector(data)
+            detections = postprocesser(cmap, paf)
+            detector_time = time.time()
+
+            tracked_objects = tracker.update(
+                detections=detections, period=frame_skip_period
+            )
+            tracker_time = time.time()
+
+            tracker_time_mean += tracker_time - detector_time
+            detector_time_mean += detector_time - start_frame_time
+        else:
+            start_frame_time = time.time()
+            tracked_objects = tracker.update()
+            tracker_time = time.time()
+            tracker_time_mean += tracker_time - start_frame_time
+
+        if create_video:
+            norfair.draw_tracked_objects(frame, tracked_objects)
+            video.write(frame)
+
+    total_time = time.time() - start_time
+    fps = total_frames / total_time
+    tracker_time_mean /= total_frames
+    detector_time_mean /= total_frames
+    return fps, tracker_time_mean, detector_time_mean
+
+
+if __name__ == "__main__":
+
+    # Flags
+    parser = argparse.ArgumentParser(description="Track human poses in a video and measure elapsed times.")
```

Comment:
The phrase "measure elapsed times" is not very clear about intention, something like "profile varios trackers and distance functions" would be clearer